pkg/integrations/dnsmasq_exporter/dnsmasq_exporter.go

```diff
@@ -2,80 +2,52 @@
 	"github.com/miekg/dns"
-	"github.com/prometheus/client_golang/prometheus"
-	"github.com/prometheus/client_golang/prometheus/promhttp"
 )
 
-// Integration is the dnsmasq_exporter integration. The integration scrapes metrics
-// from a dnsmasq server.
-type Integration struct {
-	c        Config
-	logger   log.Logger
-	exporter *collector.Collector
-}
-
-// New creates a new dnsmasq_exporter integration.
-func New(log log.Logger, c Config) (*Integration, error) {
-	exporter := collector.New(&dns.Client{
-		SingleInflight: true,
-	}, c.DnsmasqAddress, c.LeasesPath)
-
-	return &Integration{
-		c:        c,
-		logger:   log,
-		exporter: exporter,
-	}, nil
+// DefaultConfig is the default config for dnsmasq_exporter.
+var DefaultConfig Config = Config{
+	DnsmasqAddress: "localhost:53",
+	LeasesPath:     "/var/lib/misc/dnsmasq.leases",
 }
 
-// CommonConfig satisfies Integration.CommonConfig.
-func (i *Integration) CommonConfig() config.Common { return i.c.CommonConfig }
+// Config controls the dnsmasq_exporter integration.
+type Config struct {
+	// Enabled enables the integration.
+	Enabled bool `yaml:"enabled"`
 
-// Name satisfies Integration.Name.
-func (i *Integration) Name() string { return "dnsmasq_exporter" }
+	CommonConfig config.Common `yaml:",inline"`
 
-// RegisterRoutes satisfies Integration.RegisterRoutes. The mux.Router provided
-// here is expected to be a subrouter, where all registered paths will be
-// registered within that subroute.
-func (i *Integration) RegisterRoutes(r *mux.Router) error {
-	handler, err := i.handler()
-	if err != nil {
-		return err
-	}
+	// DnsmasqAddress is the address of the dnsmasq server (host:port).
+	DnsmasqAddress string `yaml:"dnsmasq_address"`
 
-	r.Handle("/metrics", handler)
-	return nil
+	// Path to the dnsmasq leases file.
+	LeasesPath string `yaml:"leases_path"`
 }
 
-func (i *Integration) handler() (http.Handler, error) {
-	r := prometheus.NewRegistry()
-	if err := r.Register(i.exporter); err != nil {
-		return nil, fmt.Errorf("couldn't register dnsmasq_exporter collector: %w", err)
-	}
-	handler := promhttp.HandlerFor(r, promhttp.HandlerOpts{})
-	return handler, nil
-}
+// UnmarshalYAML implements yaml.Unmarshaler for Config.
+func (c *Config) UnmarshalYAML(unmarshal func(interface{}) error) error {
+	*c = DefaultConfig
 
-// ScrapeConfigs satisfies Integration.ScrapeConfigs.
-func (i *Integration) ScrapeConfigs() []config.ScrapeConfig {
-	return []config.ScrapeConfig{{
-		JobName:     i.Name(),
-		MetricsPath: "/metrics",
-	}}
+	type plain Config
+	return unmarshal((*plain)(c))
 }
 
-// Run satisfies Integration.Run.
-func (i *Integration) Run(ctx context.Context) error {
-	// We don't need to do anything here, so we can just wait for the context to
-	// finish.
-	<-ctx.Done()
-	return ctx.Err()
+// New creates a new dnsmasq_exporter integration. The integration scrapes metrics
+// from a dnsmasq server.
+func New(log log.Logger, c Config) (*common.CollectorIntegration, error) {
```

Comment:
Returning `common.CollectorIntegration` feels a little weird. Maybe the integration interface should be defined in the common package and we can return the interface instead?